src/execution_plan/ops/op.h

```diff
@@ -103,6 +106,22 @@ void OpBase_AddVolatileRecord(OpBase *op, const Record r);
  * may be released. */
 void OpBase_RemoveVolatileRecords(OpBase *op);
 
+/* Mark alias as being modified by operation.
+ * Returns the ID associated with alias. */
+int OpBase_Modifies(OpBase *op, const char *alias);
+
+/* Mark alias as being projected by operation.
+ * Returns the ID associated with alias. */
+int OpBase_Projects(OpBase *op, const char *alias);
+
+/* Returns true if op is aware of alias.
+ * an operation is aware of all aliases it modifies and all aliases
+ * modified by prior operation within its segment. */
+bool OpBase_Aware(OpBase *op, const char *alias, int *idx);
```

Comment:
I believe the comment statement is only valid during execution plan construction time,
when the execution plan is fully constructed I believe this statement is false.

Reply:
This is only called within NewOp routines currently! Theoretically it could be called within Init routines as well (as `OpBase_Modifies` is for purposes), but I think that would be valid as well - the mappings are scoped even post-construction.